### DISCUSSION TITLE
Hybrid usage Langfuse among OpenAI calls as well Langchain

### DISCUSSION THREAD

**Discussion by mikeon** [langfuse/langfuse#1429] [Support] [answered]
Hi, in the docs we can find eighter  OpenAI approach when we should import langfuse.openai and use openai.chat.completions.create.... and also LangChain expression language or by handler we can pass an object     langfuse_handler = trace.get_langchain_handler()
to the e.g. retrieving     docs = retriever.get_relevant_documents(query=query, callbacks=[langfuse_handler])

But my issue is, that I don't want to use LangChain for talking to OpenAI because it has some significant problems if you rely on functions invoking. So partially I would like to use langfuse as handlers for Langchain but for some other actions don't. How to make them both working in one script.

**Comment by marcklingen** [comment]
Have you already implemented this and wonder how to best trace both with Langfuse?

**Reply by mikeon** [reply]
Yes, exactly, how to use Langfuse with both LangChain and OpenAI in the same time in the one script. Like part of retrieving docs from vectordb using handlers, but in the same time talking to openai with langfuse.openai, and make them into one log.

**Reply by marcklingen** [reply, marked as answer]
Have a look at these two examples that demonstrate interoperability of Langfuse SDKs with both integrations
- https://langfuse.com/docs/integrations/langchain/tracing#interoperability
- https://langfuse.com/docs/integrations/openai/get-started#interoperability-with-langfuse-python-sdk
